strawberryfields/gbs/vibronic.py

```diff
@@ -0,0 +1,225 @@
+the type of molecular information that should be used as input for computing molecular vibronic
+spectra with Strawberry Fields. Theoretical background on computing vibronic spectra using GBS
+can be found :ref:`here <>`.
+
+According to the Franck-Condon approximation, the probability of an electronically allowed
+vibronic transition is given by a Franck-Condon factor as
+
+.. math::
+    FCF = \left | \left \langle \phi' | \phi \right \rangle \right | ^ 2,
+
+where :math:`|\phi\rangle` and :math:`|{\phi}\rangle'` are the vibrational wave functions of the
+ground and excited electronic states, respectively. The vibrational quanta states corresponding to
+these ground and excited states, :math:`|\mathbf{n}\rangle` and :math:`|\mathbf{m'}\rangle`
+respectively, are related to each other via a Doktorov transformation as
+:math:`|\mathbf{m'}\rangle = U_{Dok} |\mathbf{n}\rangle` where :math:`U_{Dok}` is known as the
+Doktorov operator. These vibrational quanta states can also be constructed from their
+corresponding vacuum states by applying creation operators. Starting from the ground state of a
+molecule at 0 K, which corresponds to the vacuum of ground state vibrational excitations
+:math:`| \mathbf{0} \rangle`, the Franck-Condon factors can be written as:
+
+.. math::
+    FCF = \left | \left \langle \mathbf{m} |U_{Dok}| \mathbf{0} \right \rangle \right | ^ 2,
+
+where the sate :math:`| \mathbf{m} \rangle = | m_1,...m_k \rangle` contains information to obtain
+the transition energy that corresponds to the FCF through counting the number of vibrational quanta
+:math:`m` in :math:`k` modes. This is equivalent to the probability of observing output patterns
+of a GBS apparatus constructed according to :math:`U_{Dok}`. The Doktorov operator contains
+parameters that should be obtained from quantum chemistry calculations for a given molecule.
+Designing a GBS device with such molecular parameters allows obtaining the Franck-Condon profile
+of the molecule which can be used to construct the molecular vibronic spectra.
+
+The Doktorov operator can be written in terms of displacement (:math:`D_\alpha`), squeezing
+(:math:`\Sigma`) and interferometer (:math:`C_L, C_R^t`) operators as
+
+.. math::
+    U_{Dok} = C_L \Sigma C_R^t D_\alpha.
+
+The squeezing and rotation operators are obtained by singular value decomposition of a matrix
+:math:`J` as
+
+.. math::
+    J = C_L \Sigma C_R^t.
+
+In order to obtain :math:`J` we need to construct diagonal matrices :math:`\Omega` and
+:math:`\Omega'` from square roots of the ground state and excited state normal mode frequencies,
+:math:`\omega` and :math:`\omega'`, respectively. Then :math:`J` is obtained as
+
+.. math::
+    J = \Omega' U \Omega^{-1},
+
+where
+
+.. math::
+    \Omega = diag (\sqrt{\omega_1},...,\sqrt{\omega_k})
+
+    \Omega' = diag (\sqrt{\omega_1'},...,\sqrt{\omega_k'})
+
+and :math:`U` is a molecular coordinate transformation known as the Duschinsky matrix which is
```

Comment:
... known as the Duschinsky matrix which relates the normal mode coordinates of the ground and...